DFC.ServiceTaxonomy.Slack/SlackMessagePublisher.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace DFC.ServiceTaxonomy.Slack
+{
+    public class SlackMessagePublisher : ISlackMessagePublisher
+    {
+        private readonly SlackMessagePublishingConfiguration _config;
+        private readonly HttpClient _client;
+
+        public SlackMessagePublisher(IOptions<SlackMessagePublishingConfiguration> config, HttpClient client)
+        {
+            _client = client;
+            _config = config.Value;
+        }
+
+        public async Task SendMessageAsync(string text)
+        {
+            if ((_config.PublishToSlack ?? false) && !string.IsNullOrWhiteSpace(_config.SlackWebhookEndpoint))
```

Comment:
If publishing is enabled, but the endpoint is missing, it's probably better to throw an exception, rather than silently not publish.